packages/techdocs-common/src/stages/publish/awsS3.ts

```diff
@@ -142,12 +145,12 @@ export class AwsS3Publish implements PublisherBase {
         const entityRootDir = `${entity.metadata.namespace}/${entity.kind}/${entity.metadata.name}`;
         const destination = `${entityRootDir}/${relativeFilePath}`; // S3 Bucket file relative path
 
-        const fileContent = await fs.readFile(filePath, 'utf8');
+        const fileStream = await fs.createReadStream(filePath);
```

Comment:
This also prevents buffering the entire file contents in memory.

Reply:
Nice!

Reply:
except that it seems to break the tests 😢

Reply:
The AWS library is mocked inside `techdocs-common/__mocks__`. I think we need to update that as well for the tests. 😄 

https://github.com/backstage/backstage/pull/3953/files